### PR TITLE
SQL: Fix timezone issues with JDBC getDate/getTime

### LINKED ISSUE #40289: SQL: CI failure for JdbcResultSetIT.testGettingDateWithoutCalendar
https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+6.6+intake/338/console

This is reproducible in 6.6 only: `gradlew :x-pack:plugin:sql:qa:single-node:integTestRunner -Dtests.seed=9F4601A0B84D0E0E -Dtests.class=org.elasticsearch.xpack.sql.qa.single_node.JdbcResultSetIT -Dtests.method="testGettingDateWithoutCalendar" -Dtests.security.manager=true -Dtests.locale=fr-FR -Dtests.timezone=America/Costa_Rica -Dcompiler.java=11 -Druntime.java=8`

```
  2> REPRODUCE WITH: gradlew :x-pack:plugin:sql:qa:single-node:integTestRunner -Dtests.seed=9F4601A0B84D0E0E -Dtests.class=org.elasticsearch.xpack.sql.qa.single_node.JdbcResultSetIT -Dtests.method="testGettingDateWithoutCalendar" -Dtests.security.manager=true -Dtests.locale=fr-FR -Dtests.timezone=America/Costa_Rica -Dcompiler.java=11 -Dru

FAILURE 3.02s | JdbcResultSetIT.testGettingDateWithoutCalendar <<< FAILURES!
   > Throwable #1: java.lang.AssertionError: expected: java.sql.Date<071-09-07> but was: java.sql.Date<071-09-07>
   >    at __randomizedtesting.SeedInfo.seed([9F4601A0B84D0E0E:A62FAE9BFAB593CF]:0)
   >    at org.elasticsearch.xpack.sql.qa.jdbc.ResultSetTestCase.lambda$testGettingDateWithoutCalendar$118(ResultSetTestCase.java:833)
   >    at org.elasticsearch.xpack.sql.qa.jdbc.ResultSetTestCase.doWithQuery(ResultSetTestCase.java:1312)
   >    at org.elasticsearch.xpack.sql.qa.jdbc.ResultSetTestCase.testGettingDateWithoutCalendar(ResultSetTestCase.java:821)
   >    at java.lang.Thread.run(Thread.java:748)
  2> NOTE: leaving temporary files on disk at: W:\Eclipse_workspace_6.6\elasticsearch\x-pack\plugin\sql\qa\single-node\build\testrun\integTestRunner\J0\temp\org.elasticsearch.xpack.sql.qa.single_node.JdbcResultSetIT_9F4601A0B84D0E0E-001
  2> NOTE: test params are: codec=Asserting(Lucene70): {}, docValues:{}, maxPointsInLeafNode=562, maxMBSortInHeap=6.399105159024325, sim=RandomSimilarity(queryNorm=false): {}, locale=fr-FR, timezone=America/Costa_Rica
  2> NOTE: Windows 10 10.0 amd64/Oracle Corporation 1.8.0_172 (64-bit)/cpus=8,threads=1,free=448705328,total=514850816
  2> NOTE: All tests run in this JVM: [JdbcResultSetIT]
Completed [1/1] in 5.62s, 1 test, 1 failure <<< FAILURES!
```

Original error from CI:

```
java.lang.AssertionError: expected: java.sql.Date<ï¿½071-09-07> but was: java.sql.Date<ï¿½071-09-07>
	at __randomizedtesting.SeedInfo.seed([9F4601A0B84D0E0E:A62FAE9BFAB593CF]:0)
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:834)
	at org.junit.Assert.assertEquals(Assert.java:118)
	at org.junit.Assert.assertEquals(Assert.java:144)
	at org.elasticsearch.xpack.sql.qa.jdbc.ResultSetTestCase.lambda$testGettingDateWithoutCalendar$118(ResultSetTestCase.java:833)
	at org.elasticsearch.xpack.sql.qa.jdbc.ResultSetTestCase.doWithQuery(ResultSetTestCase.java:1312)
	at org.elasticsearch.xpack.sql.qa.jdbc.ResultSetTestCase.testGettingDateWithoutCalendar(ResultSetTestCase.java:821)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1750)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:938)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:974)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:988)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at org.apache.lucene.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:49)
	at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
	at org.apache.lucene.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:48)
	at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
	at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:368)
	at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:817)
	at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:468)
	at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:947)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:832)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:883)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:894)
	at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at org.apache.lucene.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:41)
	at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
	at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at org.apache.lucene.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
	at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
	at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
	at org.apache.lucene.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:54)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:368)
	at java.lang.Thread.run(Thread.java:748)
```

### PR DESCRIPTION
**Pull request by matriv**
Previously `getDate(int columnIdx)/getDate(String columnLabel)` and
`getTime(int columnIdx)/getTime(String columnLabel)` were using legacy
`java.util.Calendar` instead of the the `java.time.*` classes to reset
to the start of day or to set date to EPOCH respectively. This produced
different results for certain timestamps and timezones when calling
`getDate(col) `vs `getObject(col, java.sql.Date)` and for getTime(col)` vs
`getOBject(col, java.sql.Time)`.

Now only the methods (that must be implemented due to the JDBC spec)
`getDate(int columnIdx, Calendar cal)/getDate(String columnLabel, Calendar cal)`
and
`getTime(int columnIdx, Calendar cal)/getTime(String columnLabel, Calendar cal()`
are still using the `java.util.Calendar` for those conversion.

Fixes: #40289

### CODE REVIEW

File: x-pack/plugin/sql/qa/src/main/java/org/elasticsearch/xpack/sql/qa/jdbc/ResultSetTestCase.java [hunk 1]
```diff
@@ -5,6 +5,7 @@
  */
 package org.elasticsearch.xpack.sql.qa.jdbc;
 
+import com.carrotsearch.randomizedtesting.annotations.Repeat;
```
Comment (costin) [inline x-pack/plugin/sql/qa/src/main/java/org/elasticsearch/xpack/sql/qa/jdbc/ResultSetTestCase.java]: This doesn't seem to be used anywhere...

**Review by costin — APPROVED** [review]
LGTM

**Review by astefan — APPROVED** [review]
LGTM

File: x-pack/plugin/sql/qa/src/main/java/org/elasticsearch/xpack/sql/qa/jdbc/ResultSetTestCase.java (hunk 1 above)
Comment (matriv) [reply x-pack/plugin/sql/qa/src/main/java/org/elasticsearch/xpack/sql/qa/jdbc/ResultSetTestCase.java]: Yep, thx, leftover, removed it already

**Comment by matriv** [comment]
@elasticmachine run elasticsearch-ci/bwc

**Comment by matriv** [comment]
@elasticmachine run elasticsearch-ci/1 
@elasticmachine run elasticsearch-ci/2
@elasticmachine run elasticsearch-ci/bwc
@elasticmachine run elasticsearch-ci/default-distro
@elasticmachine run elasticsearch-ci/packaging-sample

**Pull request merged**